crates/task-impls/src/quorum_vote.rs

```diff
@@ -143,53 +306,38 @@ impl<TYPES: NodeType, I: NodeImplementation<TYPES> + 'static> HandleDepOutput
         )
         .await;
 
-        if !self.quorum_membership.has_stake(&self.public_key) {
-            debug!(
-                "We were not chosen for quorum committee on {:?}",
+        let Some(vid_share) = vid_share else {
+            error!(
+                "We don't have the VID share for this view {:?}, but we should, because the vote dependencies have completed.",
                 self.view_number
             );
             return;
-        }
+        };
 
-        // Create and send the vote.
         let Some(leaf) = leaf else {
             error!(
                 "We don't have the leaf for this view {:?}, but we should, because the vote dependencies have completed.",
                 self.view_number
             );
             return;
         };
-        let message = if let Ok(vote) = QuorumVote::<TYPES>::create_signed_vote(
-            QuorumData {
-                leaf_commit: leaf.commit(),
-            },
-            self.view_number,
-            &self.public_key,
-            &self.private_key,
-        ) {
-            GeneralConsensusMessage::<TYPES>::Vote(vote)
-        } else {
-            error!("Unable to sign quorum vote!");
+
+        let Some(cur_proposal) = cur_proposal else {
+            error!("We don't have the Quorum Proposal for this view {:?}, but we should, ecause the vote dependencies have completed.", self.view_number);
             return;
         };
```

Comment:
Why is cur_proposal even optional here?  We have the deps to vote, that must include the proposal we want to vote on.  Maybe not an issue for this PR but that seems off.

Reply:
See above.